src/gui/widgets/TimeDisplayWidget.cpp

```diff
@@ -106,12 +106,13 @@ void TimeDisplayWidget::updateTime()
 			break;
 
 		case BarsTicks:
-			m_majorLCD.setValue( s->getTacts() + 1 );
-			m_minorLCD.setValue( ( s->getTicks() % s->ticksPerTact() ) / 
-					     ( s->ticksPerTact() / s->getTimeSigModel().getNumerator() ) +1 );
-;
-			m_milliSecondsLCD.setValue( ( s->getTicks() % s->ticksPerTact() ) %
-						    ( s->ticksPerTact() / s->getTimeSigModel().getNumerator() ) );
+			int tick;
+			tick = ( s->getMilliseconds() * s->getTempo() * 48 ) / 60000 ;
```

Comment:
Yeah.  It seems like when we're doing on-the-fly math calculations in regards to time we may want to centralize them?  I'm not sure, just posing the question. :)